lib/jni/LogManager_jni.cpp

```diff
@@ -1675,3 +1681,62 @@ Java_com_microsoft_applications_events_LogManagerProvider_00024LogManagerImpl_na
         logManager->EndActivity();
     }
 }
+
+extern "C"
+JNIEXPORT jboolean JNICALL
+Java_com_microsoft_applications_events_LogManagerProvider_00024LogManagerImpl_nativeRegisterSubstrateSignals(
+        JNIEnv *env, jobject thiz, jlong native_log_manager) {
+#if HAS_SS
+    auto logManager = getLogManager(native_log_manager);
+    auto ss = SubstrateSignalsHelper::GetSubstrateSignalsInspector();
+    if(ss != nullptr) {
+        logManager->SetDataInspector(ss);
+        return true;
+    }
+#endif
+    return false;
+}
+extern "C"
+JNIEXPORT jboolean JNICALL
+Java_com_microsoft_applications_events_LogManagerProvider_00024LogManagerImpl_nativeUnregisterSubstrateSignals(
+        JNIEnv *env, jobject thiz, jlong native_log_manager) {
+#if HAS_SS
+    // TODO: implement nativeUnregisterSubstrateSignals()
+    auto logManager = getLogManager(native_log_manager);
+    auto ss = SubstrateSignalsHelper::GetSubstrateSignalsInspector();
+    if(ss != nullptr) {
+        logManager->RemoveDataInspector(ss->GetName());
+        return true;
+    }
+#endif
+    return false;
+}
+
+extern "C"
+JNIEXPORT jboolean JNICALL
+Java_com_microsoft_applications_events_LogManager_nativeRegisterSubstrateSignalsOnDefaultLogManager(JNIEnv *env, jclass clazz) {
```

Comment:
nit - we can move the `LogManager_*` functions as part of the single encapsulated `extern "C"` to be in consistent with the existing pattern.